geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/formats/LeafletMapExporter.scala

```diff
@@ -0,0 +1,189 @@
+  val indexFile: File = getDestination(Option(params.file).getOrElse{
+    new File(new File(GEOMESA_HOME.get), "leaflet")
+  })
+  val indexWriter: Writer = ExportCommand.getWriter(indexFile, null)
+
+  val (indexHead, indexTail): (String, String) = {
+    val indexStream: InputStream = getClass.getClassLoader.getResourceAsStream("leaflet/index.html")
+    try {
+      val indexString: String = Source.fromInputStream(indexStream).mkString
+      val indexArray: Array[String] = indexString.split("\\|codegen\\|")
+      require(indexArray.length == 2, "Malformed index.html unable to render map.")
+      (indexArray(0), indexArray(1))
+    } finally {
+      indexStream.close()
+    }
+  }
+
+  override def start(sft: SimpleFeatureType): Unit = {
+    this.sft = sft
+    indexWriter.write(indexHead)
+    indexWriter.write("""var points = {"type":"FeatureCollection","features":[""")
+  }
+
+  override def export(features: Iterator[SimpleFeature]): Option[Long] = {
+    var count = 0L
+    features.foreach { feature =>
+      if (first) {
+        first = false
+      } else {
+        indexWriter.write(",")
+      }
+      json.writeFeature(feature, indexWriter)
+      storeFeature(feature, coordMap)
+      count += 1L
+    }
+    indexWriter.flush()
+    totalCount += count
+    Some(count)
+  }
+
+  override def close(): Unit  = {
+    // Finish writing GeoJson
+    indexWriter.write("]};\n\n")
+    indexWriter.write(getFeatureInfo(sft))
+    // Write Heatmap Data
+    val values = normalizeValues(coordMap)
+    first = true
+    indexWriter.write("""var heat = L.heatLayer([""" + "\n")
+    values.foreach{ c =>
+      if (first) {
+        first = false
+      } else {
+        indexWriter.write(",\n")
+      }
+      indexWriter.write(c._1.render(c._2))
+    }
+    indexWriter.write("\n    ], {radius: 25});\n\n")
+    indexWriter.write(indexTail)
+    indexWriter.close()
+
+    if (totalCount < 1) user.warn("No features were exported. " +
+      "This will cause the map to fail to render correctly.")
+    // Use println to ensure we write the destination to stdout so the bash wrapper can pick it up.
+    System.out.println("\nSuccessfully wrote Leaflet html to: " + indexFile.toString + "\n")
+  }
+}
+
+object LeafletMapExporter {
+  def getDestination(file: File): File = {
+    implicit class WritableFile(file: File) {
+      def ensureWritable: File = {
+        if (file.canWrite) { file }
+        else { throw new SecurityException(s"Unable to create output destination ${file.toString}, check permissions.") }
+      }
+    }
+
+    // Handle both files and directories that could exist or not
+    if (file.exists()) {
+      if (file.isDirectory) {
+        new File(file, "index.html").ensureWritable
+      } else {
+        if (file.toString.endsWith(".html")) { file.ensureWritable }
+        else { throw new RuntimeException(s"Destination file ${file.toString} must end with '.html'") }
+      }
+    } else {
+      file.mkdir()
```

Comment:
might want to switch this to `.mkdirs()` which will create nested folders if required